### DISCUSSION TITLE
403 Forbidden

### DISCUSSION THREAD

**Discussion by SamuelHadsall** [solspace/craft-freeform#352] [Questions & Answers]
I have a freeform form built with vue and submitting post using axios. I'm now getting a 403 forbidden when submitting contact form.  I think it has to do with an update with Freeform and noticed the anonymous access to the API endpoint was removed. Do I need to pass creds with my post submission

```
await this.$axios.$post('/form', formData, {            
            headers: {
              // For CraftCMS to detect isAjax properly.
              'X-Requested-With': 'XMLHttpRequest',
            },            
          })`
```

Old Version
![apicontrolleroldversion](https://user-images.githubusercontent.com/6413981/169137936-74ead711-1f3e-4475-bd5f-41758ce57d14.png)

New Version
![apicontrollernewversion](https://user-images.githubusercontent.com/6413981/169137955-8bb929f6-48da-4a67-8faf-6006a8d98425.png)

**Comment by kjmartens** [comment]
Hi @SamuelHadsall,

Can you confirm what you have set for the "Freeform Session Context" setting inside the Freeform Form Behavior settings page? If it's PHP sessions, can you switch to Encrypted Payload and see if that helps? Or if it's Encrypted Payload, does switching to PHP sessions help?
https://docs.solspace.com/craft/freeform/v3/setup/settings.html#freeform-session-context
​
If using PHP Sessions for the Freeform Session Context setting, please be sure to include `{% do form.registerContext %}` below your form. More info here:
https://docs.solspace.com/craft/freeform/v3/templating/caching.html
​
If using Encrypted Payload, add `freeform_payload` to the request with the form instance's `json.freeform_payload` property. Here's some further reading for Vue implementations (including how to add payload):
https://docs.solspace.com/craft/freeform/v3/guides/using-vue-js.html

**Reply by SamuelHadsall** [reply]
Oh Dang! Thank you for replying so fast!

We are using Encrypted Payload and I switched to PHP Session, but that did not resolve the issue. I am not however using the freeform_payload in the request. I did add that

```
 const formData = {
          freeform_payload: this.form.freeform_payload,
          formHash: this.form.hash,          
          [this.$csrf.name]: this.$csrf.value,
          ...this.form.data,
        }
```
But that did not resolve the issue:
```
formHash: "mgrYvwQPl-aqxMJqGRy-IMNjBw5MgoVXqBI80BxIYGrIwcIHfQwQZMRXzGnV",…}
CRAFT_CSRF_TOKEN: "mGSWafiXLl5Fff3nxwvFvFZVU1x2OpON2W1wwPCL-KDeVQlEUnxvQdBd3Fme5xYENTTJpLRSqvsdP2UpW1iltOBaKvfdxrTI5hZzNDMXWXM="
address: "Test Drive."
city: "Wentzville"
email: "shadsall@steadyrain.com"
firstName: "Test"
formHash: "mgrYvwQPl-aqxMJqGRy-IMNjBw5MgoVXqBI80BxIYGrIwcIHfQwQZMRXzGnV"
lastName: "Tester"
message: "Testing"
phone: "3093314509"
state: "MO"
zipCode: "63885"
```
